pkgs/applications/editors/emacs/build-support/melpa.nix

```diff
@@ -62,24 +64,26 @@ in
     recipe: Optional MELPA recipe.
     Default: a minimally functional recipe
   */
-, recipe ? (writeText "${pname}-recipe" ''
-    (${ename} :fetcher git :url ""
-              ${lib.optionalString (files != null) ":files ${files}"})
+, recipe ? (writeText "${finalAttrs.pname}-recipe" ''
```

Comment:
Not something to address here and now:
Do we really need to use `writeText` for the recipe? Couldn't it be done inline in the builder instead, skipping the intermediate derivation?

Reply:
Sounds good.  I created https://github.com/NixOS/nixpkgs/issues/334888 to track this potential improvement.